force-app/main/default/classes/AnotherOpportunityHandler.cls

```diff
@@ -0,0 +1,133 @@
+public with sharing class AnotherOpportunityHandler {
+    public static void setOppType(List<Opportunity> newOpps) {
+        // Set default Type for new Opportunities
+        for (Opportunity opp : newOpps) {
+            if (opp.Type == null) {
+                opp.Type = 'New Customer';
+        }
+    }
+}
+    public static void preventOppDeletion(List<Opportunity> oldOpps) {
+        // Prevent deletion of closed Opportunities
+
+        for (Opportunity opp : oldOpps){
+            // Debugging StageName and Opportunity details
+            System.debug('$$$Checking Opportunity for deletion: ' + opp.Name + ' | Stage: ' + opp.StageName);
+            if (opp.StageName == 'Closed Won' || opp.StageName == 'Closed Lost') {
```

Comment:
Try to remember to check the industry also. This can also be done with the addError